include/arch/arm/cortex_m/mpu/arm_mpu_v7m.h

```diff
@@ -159,7 +188,46 @@ typedef struct arm_mpu_region_attr arm_mpu_region_attr_t;
 		__is_writable__; \
 	})
 
+/*
+ * @brief Evaluate Execution allowance
+ *
+ * Evaluate whether the access permissions include execution.
+ *
+ * @param attr The k_mem_partition_attr_t object holding the
+ *             MPU attributes to be checked against execution
+ *             allowance.
+ */
 #define K_MEM_PARTITION_IS_EXECUTABLE(attr) \
-	(!((attr) & (NOT_EXEC)))
+	(!((attr.rasr_attr) & (NOT_EXEC)))
+
+/* Attributes for no-cache enabling */
+
+#define K_MEM_PARTITION_P_NA_U_NA_NOCACHE {(K_MEM_PARTITION_P_NA_U_NA \
```

Comment:
@aurel32 FYI
Could you , please, review the No-Cache related attributes which are now introduced in ARMv7, v8 as K_MEM_PARTITION_xxxx macros?

Reply:
Right, i missed that; thanks for catching this. I will update and update also in the ARMv8-M, so they are aligned with each other.